### PR TITLE
Forms feature

### PR DESCRIPTION
**Pull request by mariobarrioss**
All the features required in the assignment were added.

### CODE REVIEW

File: app/views/users/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html>
+
+<html>
```
Comment (maelfosso) [inline app/views/users/new.html.erb]: Not needed.
Read that for a better understanding https://guides.rubyonrails.org/layouts_and_rendering.html

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        <%= f.submit "Create My Account", class:"btn" %>
+    <% end %>-->
+
+    <%= form_for @user do |f| %>
+        Username: <%= f.text_field :username, placeholder: "Your user name here" %><br />
+        Email: <%= f.text_field :email, placeholder: "example@example.com" %><br />
+        Password: <%= f.password_field :password, placeholder: "Your password here" %><br />
+        <%= f.submit "Create My Account", class:"btn" %>
+    <% end %>
+</html>
```
Comment (maelfosso) [inline app/views/users/new.html.erb]: Not needed

File: app/views/users/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    
+    <!--<%= form_tag("/users", method: "post") do %>
+        <%= label_tag(:username, "Username:") %>
+        <%= text_field_tag(:username) %>
+        <%= label_tag(:email, "Email:") %>
+        <%= text_field_tag(:email) %>
+        <%= label_tag(:password, "Password:") %>
+        <%= password_field_tag(:password) %>
+        <%= submit_tag("Submit") %>
+    <% end %>-->
```
Comment (maelfosso) [inline app/views/users/new.html.erb]: Please use `user_path` instead of `/users`

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        if @user.update(users_params)
+            redirect_to edit_user_path
+        else
+            render :edit
+        end
+    end
+
+    def create
+        #Manual form entry
+        #@user = User.new(:username => params[:username], :email => params[:email], :password => params[:password])
```
Comment (maelfosso) [inline app/controllers/users_controller.rb]: Remove the comment please

**Review by maelfosso — CHANGES_REQUESTED** [review]
# CHanges required :red_circle: 

Hello,
Nice code !!! However, there are some bugs that need to be fixed

1.
```ruby
Started GET "/users/3/edit" for ::1 at 2020-03-06 03:06:22 -0500
   (0.1ms)  SELECT sqlite_version(*)
Processing by UsersController#edit as HTML
  Parameters: {"id"=>"3"}
  User Load (0.3ms)  SELECT "users".* FROM "users" WHERE (3) LIMIT ?  [["LIMIT", 1]]
  ↳ app/controllers/users_controller.rb:7:in `edit'
Return value is: nil

[4, 13] in /home/maelfosso/Documents/Microverse/TSE/re-former/app/controllers/users_controller.rb
    4:     end
    5: 
    6:     def edit
    7:         @user = User.find_by(params[:id])
    8:         byebug
=>  9:     end
   10: 
   11:     def update
   12:         @user = User.find_by(params[:id])
   13:         byebug
(byebug) @user
#<User id: 1, username: "alice bar", email: "alice@bar.com", password: nil, created_at: "2020-03-06 08:01:22", updated_at: "2020-03-06 08:02:05", password_digest: [FILTERED]>
(byebug) params
<ActionController::Parameters {"controller"=>"users", "action"=>"edit", "id"=>"3"} permitted: false>
(byebug) 
```
While we have this in the database
```ruby
2.6.0 :001 > User.all
   (1.1ms)  SELECT sqlite_version(*)
  User Load (0.3ms)  SELECT "users".* FROM "users" LIMIT ?  [["LIMIT", 11]]
 => #<ActiveRecord::Relation [#<User id: 1, username: "alice bar", email: "alice@bar.com", password: nil, created_at: "2020-03-06 08:01:22", updated_at: "2020-03-06 08:02:05", password_digest: [FILTERED]>, #<User id: 2, username: "Bob", email: "bob@bar.com", password: nil, created_at: "2020-03-06 08:01:31", updated_at: "2020-03-06 08:01:31", password_digest: [FILTERED]>, #<User id: 3, username: "Foo", email: "foo@bar.com", password: nil, created_at: "2020-03-06 08:05:25", updated_at: "2020-03-06 08:05:25", password_digest: [FILTERED]>]> 
2.6.0 :002 > 
```

2. Please, read the comments below

Kindly fix the bugs and make the suggested changes and submit a new code review request
Thanks

Mael FOSSO

**Comment by rsundar** [comment]
# CHANGES MADE

- fixed the bug found by @maelfosso,
- Removed extraneous comments,
- Removed ```<DOCTYPE html>``` tag form views.

File: app/controllers/users_controller.rb (hunk 4 above)
Comment (rsundar) [reply app/controllers/users_controller.rb]: Done

File: app/views/users/new.html.erb (hunk 1 above)
Comment (rsundar) [reply app/views/users/new.html.erb]: DONE

File: app/views/users/new.html.erb (hunk 3 above)
Comment (rsundar) [reply app/views/users/new.html.erb]: Done

File: app/views/users/new.html.erb (hunk 2 above)
Comment (rsundar) [reply app/views/users/new.html.erb]: removed

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    def update
+        if @user.update(users_params)
+            redirect_to edit_user_path
+        else
+            render :edit
+        end
+    end
+
+    def create
+        #@user = User.new(:username => params[:username], :email => params[:email], :password => params[:password])
```
Comment (wathigo) [inline app/controllers/users_controller.rb]: **[REQUIRED]**
Let's remove commented code([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from Pathwright)

File: app/views/users/new.html.erb (hunk 1 above)
Comment (wathigo) [inline app/views/users/new.html.erb]: As mentioned by @maelfosso  the `html` tags are not necessary. We already have them in `app/views/layouts/application.html.erb` 
Let's get rid of them please.

File: app/views/users/new.html.erb (hunk 2 above)
Comment (wathigo) [inline app/views/users/new.html.erb]: Let's remove this one too

**Review by wathigo — CHANGES_REQUESTED** [review]
### Changes Required
Great work on fixing the bug :clap: 
Let's fix just a few issues before we proceed to the next project.
See the comments :point_down: 

**After implementing the changes, submit another request for review.**

**Happy Coding :computer: 
Simon Wathigo**

**Comment by rsundar** [comment]
# CHANGES MADE

- removed ```<!DOCTYPE html>``` and ```<html>``` tags as requested.
- removed commented out code.

File: app/views/users/new.html.erb (hunk 1 above)
Comment (rsundar) [reply app/views/users/new.html.erb]: done

File: app/views/users/new.html.erb (hunk 2 above)
Comment (rsundar) [reply app/views/users/new.html.erb]: done

File: app/controllers/users_controller.rb (hunk 5 above)
Comment (rsundar) [reply app/controllers/users_controller.rb]: Done

**Review by wathigo — APPROVED** [review]
Approved :heavy_check_mark: 
Great work implementing the requested changes :clap: 

**You may move on.**

**Good luck :four_leaf_clover: 
Simon Wathigo**

**Comment by rsundar** [comment]
Thanks for the excellent feedback @wathigo @maelfosso :heart:

**Pull request merged**